contracts/util/ERC20/UnlimitedSupplyToken.sol

```diff
@@ -12,23 +12,61 @@ pragma solidity ^0.7.4;
 import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
 
 contract UnlimitedSupplyToken is ERC20 {
-    address public immutable unlimited;
+    address public admin;
+    mapping(address => bool) public minterMap;
 
     constructor(
         string memory _name,
         string memory _symbol,
-        address _unlimited
+        address[] memory _minters,
+        address _admin
     ) ERC20(_name, _symbol) {
-        unlimited = _unlimited;
+        require(_admin != address(0), 'Invalid Admin Address');
+        admin = _admin;
+
+        for (uint256 i = 0; i < _minters.length; ++i) {
+            require(_minters[i] != address(0), 'Invalid Minter Address');
+            minterMap[_minters[i]] = true;
+        }
+    }
+
+    modifier mintable() {
+        require(minterMap[msg.sender] == true, "You don't have mint right");
+        _;
+    }
+
+    modifier modifiable() {
```

Comment:
@GarfDev Please rename to `onlyAdmin()` since we have a similar naming pattern for these kind of modifiers (onlyOwner for example).